common/common.cpp

```diff
@@ -1340,8 +1340,8 @@ std::string llama_detokenize_bpe(llama_context * ctx, const std::vector<llama_to
     std::string piece;
     std::string result;
 
-    for (size_t i = 0; i < tokens.size(); ++i) {
-        piece = llama_token_to_piece(ctx, tokens[i]);
+    for (int token : tokens) {
```

Comment:
What is the point of this change?
I personally vastly prefer this version to the original. But this project coding style in the C++ parts of it is very C-like, so I have never dared to go and change the style that I dislike.

Reply:
This exact place was fine as it was, but a lot of other for-loops had signed/unsigned mismatches and/or were followed by creating a reference to the i-th object.
So it's less about styling and more about making the code shorter and more expressive.